tests/unit/preprocessors/test_converted_notebook_preprocessor.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+import tarfile
+import os
+import posixpath
+from pathlib import Path
+
+import fairing
+from fairing.preprocessors.converted_notebook import ConvertNotebookPreprocessor
+from fairing.constants.constants import DEFAULT_DEST_PREFIX
+
+NOTEBOOK_PATH = os.path.relpath(
+    os.path.join(os.path.dirname(__file__), 'test_notebook.ipynb'))
+CONVERTED_NOTEBOOK_PATH = NOTEBOOK_PATH.replace(".ipynb",".py")
+
+def test_preprocess():
+    preprocessor = ConvertNotebookPreprocessor(notebook_file=NOTEBOOK_PATH)
+    files = preprocessor.preprocess()
+    assert Path(CONVERTED_NOTEBOOK_PATH) in files
+
+def test_get_command():
```

Comment:
Does pytest execute the tests in order? i.e. when executing this one, does it guarantee that previous tests passed or were at least ran? that way if test_preprocess fails, we can ignore errors from this test.

Reply:
No. It shouldn't by design. I think what you are asking for is a smoke test but these are unit tests so we wont incur time running all the tests so no gating mechanism is necessary. However, we can do something for integration tests where unit or some other tests failures would not run the remaining tests.